tests/codegen/range-attribute.rs

```diff
@@ -0,0 +1,33 @@
+// Checks that range metadata gets emitted on functions result and arguments
+// with scalar value.
+
+//@ compile-flags: -O -C no-prepopulate-passes
+//@ min-llvm-version: 19
+
+#![crate_type = "lib"]
+
+use std::num::NonZero;
+
+// CHECK: noundef range(i64 1, 0) i64 @nonzero_int(i64 noundef range(i64 1, 0) %x)
```

Comment:
I think you can do

```suggestion
// CHECK-LABEL: @nonzero_int
// CHECK-SAME: noundef range(i64 1, 0) i64 @nonzero_int(i64 noundef range(i64 1, 0) %x)
```

so you get the better section splitting from CHECK-LABEL

Reply:
CHECK-SAME will start matching after the CHECK-LABEL so it is not possible to verify that the result get a range applied in that case unless it is moved to the CHECK-LABEL like
`CHECK-LABEL: noundef range(i64 1, 0) i64 @nonzero_int`
and then it feels like a CHECK is better